xwiki-platform-core/xwiki-platform-mentions/xwiki-platform-mentions-default/src/main/java/org/xwiki/mentions/internal/DefaultMentionNotificationService.java

```diff
@@ -49,17 +49,32 @@
     @Inject
     private EntityReferenceSerializer<String> serializer;
 
+    @Inject
+    private QuoteService quote;
+
+    @Inject
+    private MentionsConfiguration configuration;
+
     @Override
-    public void sendNotif(DocumentReference authorReference, DocumentReference documentReference,
-        DocumentReference mentionedIdentity, MentionLocation location, String anchorId)
+    public void sendNotification(MentionNotificationParameters mentionNotificationParameters)
     {
+        String extractedQuote;
+        if (this.configuration.isQuoteActivated()) {
```

Comment:
I'm not entirely sure about that one: right now, if you had the config off and you enable it later, it means that all the already computed mentions won't have any associated quote. Only the new ones will have quotes. 
On the other hand, without this if you might compute quotes that you will never display if you don't change the config.

IMO it really depends why we have this config: if this is for perf reason, then I agree we should have this if. If it's for display reason, then we shouldn't and it should be only a switch for the velocity template.

Reply:
imo notifications are transient and consequently it it not very useful to compute a quote which will never be display during the short lifespan during which the notification might be displayed to the user.
But we can easily separate the settings, one for the generation of the quotes (regarding performances), and a second one for the quotes display (regarding UX).

Reply:
> But we can easily separate the settings, one for the generation of the quotes (regarding performances), and a second one for the quotes display (regarding UX).

A bit overkill for now. I'm ok for keeping it like that, we just need to document properly the behaviour of this option.